### PR TITLE
[macOS, Mono] Automatically enable JIT entitlements for the Mono exports.

### PR DESCRIPTION
**Pull request by bruvzg**
Automatically enable entitlements which are always required to run managed code.

### CODE REVIEW

File: platform/osx/export/export.cpp [hunk 1]
```diff
@@ -163,9 +164,11 @@ void EditorExportPlatformOSX::get_export_options(List<ExportOption> *r_options)
 	r_options->push_back(ExportOption(PropertyInfo(Variant::BOOL, "codesign/replace_existing_signature"), true));
 	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "codesign/entitlements/custom_file", PROPERTY_HINT_GLOBAL_FILE, "*.plist"), ""));
 
+#ifndef MODULE_MONO_ENABLED
```
Comment (akien-mga) [inline platform/osx/export/export.cpp:167]: I'd add an explanatory comment to make it clear why those are compiled out.

**Review by akien-mga — APPROVED** [review]

File: platform/osx/export/export.cpp [hunk 2]
```diff
@@ -41,6 +41,7 @@
 #include "editor/editor_export.h"
 #include "editor/editor_node.h"
 #include "editor/editor_settings.h"
+#include "modules/modules_enabled.gen.h"
```
Comment (akien-mga) [inline platform/osx/export/export.cpp:44]: That won't work on `3.x` which predates this refactoring:
```
platform/osx/export/export.cpp:44:10: fatal error: 'modules/modules_enabled.gen.h' file not found
#include "modules/modules_enabled.gen.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

But you can instead rely on `if (Engine::get_singleton()->has_singleton("GodotSharp"))` to check if there's C# support.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Comment by Calinou** [comment]
Should this be cherry-picked to `3.3`? See https://github.com/godotengine/godot/issues/52366.

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.4.

**Pull request merged**